fulp_modules/main_features/bloodsuckers/code/bloodsucker/bloodsucker_life.dm

```diff
@@ -263,24 +260,25 @@
 	if(!owner.current || AmFinalDeath)
 		return
 
-	/// Nutrition - The amount of blood is how full we are.
+	// Nutrition - The amount of blood is how full we are.
 	owner.current.set_nutrition(min(owner.current.blood_volume, NUTRITION_LEVEL_FED))
 
 	// BLOOD_VOLUME_GOOD: [336] - Pale
 //	handled in bloodsucker_integration.dm
+
+	// BLOOD_VOLUME_EXIT: [250] - Exit Frenzy (If in one) This is high because we want enough to kill the poor soul they feed off of.
+	if(owner.current.blood_volume >= FRENZY_THRESHOLD_EXIT && Frenzied)
+		qdel(owner.current.GetComponent(/datum/component/bloodsucker_frenzy))
```

Comment:
Wouldn't it make more sense to simply handle all frenzy code inside of the frenzy component, rather than adding and qdel-ing the component whenever it's needed?